src/hotspot/share/gc/shenandoah/shenandoahFreeSet.hpp

```diff
@@ -779,7 +785,11 @@ using idx_t = ShenandoahSimpleBitMap::idx_t;
   // locked action can be seen by these unlocked functions.
   inline size_t capacity()  const { return _partitions.capacity_of(ShenandoahFreeSetPartitionId::Mutator);             }
   inline size_t used()      const { return _partitions.used_by(ShenandoahFreeSetPartitionId::Mutator);                 }
-  inline size_t available() const { return _partitions.available_in_not_locked(ShenandoahFreeSetPartitionId::Mutator); }
+  inline size_t available() {
+    shenandoah_assert_not_heaplocked();
+    ShenandoahRebuildLocker locker(rebuild_lock());
```

Comment:
May be motivate in a brief comment why we need the rebuild lock in this API, but not around the other APIs such as capacity() and used()?